civihr_employee_portal/js/ta-documents-app.js

```diff
@@ -0,0 +1,52 @@
+angular.module('taDocuments', ['civitasks.appDocuments']).controller('ModalController', ['$scope', '$rootScope', '$rootElement', '$log', '$uibModal', 'DocumentService', 'FileService', 'config',
+  function ($scope, $rootScope, $rootElement, $log, $modal, DocumentService, FileService, config) {
+    this.modalDocument = function (id, role) {
+
+      DocumentService.get({'id': id})
+      .then(function (data) {
+        // Display the modal
+        !!data && openModalDocument(data[0], role);
+      }, function (reason) {
+        CRM.alert(reason, 'Error', 'error');
+      });
+    }
+
+    /**
+     * Opens Document Modal with or without document data
+     *
+     * @param {object} data
+     */
+    function openModalDocument (data, role) {
+      var modalInstance = $modal.open({
+        appendTo: $rootElement.find('div').eq(0),
+        templateUrl: config.path.TPL + 'modal/document.html?v=3',
+        controller: 'ModalDocumentCtrl',
+        resolve: {
+          role: function () {
+            return [role];
+          },
+          data: function () {
+            return data;
+          },
+          files: function () {
+            if (!data.id || !+data.file_count) {
+              return [];
+            }
+
+            return FileService.get(data.id, 'civicrm_activity');
+          }
+        }
+      });
+
+      modalInstance.result.then(function (results) {
+        $scope.$broadcast('documentFormSuccess', results, data);
+
+        if (results.open) {
+          $rootScope.modalDocument(data);
```

Comment:
I have taken the above code form T&A
```js
modalInstance.result.then(function (results) {
   ...
})
```
I realized, this is only needed in T&A, as there needs to identify the status of the document and should be removed form the list of documents if the document is approved or rejected and keep otherwise. And `$rootScope.modalDocument(data);` is used to display the modal  if the button "Save and New" was clicked. But now that the button is removed, we wont be needing now.